File: rb/features.rb
# ArcgisHubWorldCountriesGeneralized SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ArcgisHubWorldCountriesGeneralizedFeatures
  def self.make_feature(name)
    case name
    when "base"
      ArcgisHubWorldCountriesGeneralizedBaseFeature.new
    when "test"
      ArcgisHubWorldCountriesGeneralizedTestFeature.new
    else
      ArcgisHubWorldCountriesGeneralizedBaseFeature.new
    end
  end
end

File: py/features.py
# ArcgisHubWorldCountriesGeneralized SDK feature factory

from feature.base_feature import ArcgisHubWorldCountriesGeneralizedBaseFeature
from feature.test_feature import ArcgisHubWorldCountriesGeneralizedTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ArcgisHubWorldCountriesGeneralizedBaseFeature(),
        "test": lambda: ArcgisHubWorldCountriesGeneralizedTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
